### ISSUE TITLE
Breaking change to listCse function in 0.130

### ISSUE THREAD

**Issue by jedi58** [googleapis/google-api-php-client-services#197]
In 0.130 the `listCse` function has been changed from: `public function listCse($q, $optParams = array())` to: `public function listCse($optParams = array())` so is a breaking change with only a minor version number change

**Comment by jdpedrie** [comment]
This library contains auto-generated clients which map directly to the Google APIs. That means you can always use the latest version of the clients in your code without waiting for someone to manually update the clients, but it also means it's hard to control for breaking changes.

Outside of development, you should pin `google/apiclient-services` to a specific minor version, and not update that until you've verified that it works.

**Comment by jackbentley** [comment]
@jdpedrie Does this library use [Semver](https://semver.org/)?

This issue also seems similar to #190 (different area, same issue?)

**Comment by jdpedrie** [comment]
Yes in the technical sense that it is versioned pre-1.0, where breaking changes are allowed in minor versions.

**Comment by jackbentley** [comment]
Excellent, thanks. Would it be worth upping the version to v1 - as it does seem we've been on v0 for about 4 years now? Especially considering this library interfaces with live, stable, APIs - it's a bit misleading and doesn't infuse confidence or perceived reliability in the library?

**Comment by jdpedrie** [comment]
The advantage of pre-1.0 versions is that composer treats them like major versions.

For instance, if you require `^v0.1`, composer will install `v0.1.100000`, but not `v0.2.0`. For pre-1.0 packages, composer assumes there may be breaking changes between minor versions.

Once a library reaches 1.0, composer starts assuming no breaking changes within major versions. That means that requiring `^v1.0` would install `v1.2.0`.

In practice for this library, we'd need to bump the major version with every release, since they are generated automatically. Since we've already established the pattern of minor versions, there's no real need to change it.

**Comment by jdpedrie** [comment]
> I'm unsure if the project should be specifying both libraries or not?

Yes, that's the safest course. I opened https://github.com/googleapis/google-api-php-client/pull/1823 to make this clearer in the documentation for the API client.

> does the PHP client require the client services in order to provide it as an API or does it have an actual dependency on the code?

The former. `google/apiclient` doesn't call into `google/apiclient-services`. In this way the dependency tree is inverted.

> If my suspicions are correct, setting minimum-stability in composer to stable might prevent you from using the library entirely? 

It does not prevent installation. It would prevent you from installing `dev-master`, but not a pre-1.0 tag.

All your points are valid and are things we've been considering. The current situation is less than ideal and we're looking at ways we can practically improve it.

The proper way to solve it is to think of this repository as a snapshot of the state of Google APIs at a given point in time. Freeze that snapshot at the point when you develop against it.

Thanks for the research, you stated the issues very well. I opened #1823 to improve the documentation around this issue, and we'll keep working to improve the overall situation as well.
